### PR TITLE
import GPUNotAvailableError to error.py

### LINKED ISSUE #1649: Move `GPUNotAvailableError` to `error.py`.
### Description

In a branch off `trunk-minor`:
* Import `GPUNotAvailableError` in `error.py`.
* Note in the documentation that `util.GPUNotAvailableError` is deprecated and will move to `error`.

After this branch is merged, remove `util.GPNotavailableError` in a branch off `trunk-major.

### Motivation and context

`util` should hold utility methods and `error` should hold error classes.

### PR DESCRIPTION
**Pull request by melodyyzh**
## Description
Imported `GPUNotAvailableError` in `error.py`.

Will remove `util.GPNotavailableError` in a branch off `trunk-major` after this branch is merged

## Motivation and context

`util` should hold utility methods and `error` should hold error classes.

Resolves #1649 

## Change log
```
Added

* ``hoomd.error.GPUNotAvailableError``
  (`#1694 <https://github.com/glotzerlab/hoomd-blue/pull/1694>`__).

Deprecated:

* ``hoomd.util.GPUNotAvailableError``
  (`#1694 <https://github.com/glotzerlab/hoomd-blue/pull/1694>`__).
```

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Review by tommy-waltmann — COMMENTED** [review]
In the referenced issue (#1649), it states that this PR should also "Note in the documentation that util.GPUNotAvailableError is deprecated and will move to error."

**Review by joaander — CHANGES_REQUESTED** [review]
To complete the move in a backward compatible way:
1. Move `_NoGPU` from `util.py` to `error.py`.
2. Adjust all uses of `_NoGPU` to import it from `error`.
3. Add `GPUNotAvailableError` to `module-hoomd-errors.rst` so it appears in the documentation.
4. Remove the definition of `GPUNotAvailableError` from `util.py` and replace it with an import of `hoomd.error.GPUNotAvailableError` to preserve backward compatibility.
5. Note the deprecation in `deprecated.rst`.

Merge `trunk-minor` into `move_GPUNotAvailableError` first to avoid creating conflicts in `deprecated.rst`.

File: hoomd/error.py [hunk 1]
```diff
@@ -37,6 +37,19 @@ class TypeConversionError(ValueError):
     pass
 
 
+class GPUNotAvailableError(NotImplementedError):
+    """Error for when a GPU specific feature was requested without a GPU."""
+    pass
+
+
+class _NoGPU:
```
Comment (tcmoore3) [inline hoomd/error.py:45]: Should you remove this class from `hoomd/util.py`?

Comment (melodyyzh) [reply hoomd/error.py:45]: You are right, thanks!

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**